core/core-funcs.el

```diff
@@ -36,12 +36,18 @@
   (run-hooks 'text-mode-hook))
 
 (defun spacemacs//get-package-directory (pkg)
-  "Return the directory of PKG. Return nil if not found."
-  (let ((elpa-dir (file-name-as-directory package-user-dir)))
-    (when (file-exists-p elpa-dir)
-      (let* ((pkg-match (concat (symbol-name pkg) "-[0-9]+"))
-             (dir (car (directory-files elpa-dir 'full pkg-match))))
-        (when dir (file-name-as-directory dir))))))
+  "Return the installed directory of PKG. Return `nil' if not found."
+  (let ((directories (cons package-user-dir
+                           (and dotspacemacs-use-shared-packages
+                                package-directory-list)))
+        elpa-dir dir)
+    (while (and (not dir) directories)
+      (setq elpa-dir (pop directories))
+      (when (file-exists-p elpa-dir)
+        (let* ((pkg-match (concat (symbol-name pkg) "-[0-9]+"))
+               (dir (car (directory-files elpa-dir 'full pkg-match))))
+          (when dir (setq dir (file-name-as-directory dir))))))
+    dir))
```

Comment:
I considered using catch/throw as well.  To me importance is not whether I use "idomatic" code.
To me writing simple code that most people can understand is a very good reason to stick with
simple syntax when it does not hurt that much.  I use catch/throw when I think it improves the
readability of the code.  In this particular case, I'm not sure that it matters too much.